applications/ContactStructuralMechanicsApplication/tests/test_sparse_multiplication.py

```diff
@@ -8,14 +8,53 @@
 def GetFilePath(fileName):
     return os.path.dirname(os.path.realpath(__file__)) + "/" + fileName
 
+class TestSparseMatrixSum(KratosUnittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def __sparse_matrix_sum(self):
+        file_name = "../../../kratos/tests/A.mm"
+
+        # Read the matrices
+        A = KratosMultiphysics.CompressedMatrix()
+        B = KratosMultiphysics.CompressedMatrix()
+        KratosMultiphysics.ReadMatrixMarketMatrix(GetFilePath(file_name),A)
+        KratosMultiphysics.ReadMatrixMarketMatrix(GetFilePath(file_name),B)
+
+        try:
+            from scipy import sparse, io
+            import numpy as np
+            missing_scipy = False
+        except ImportError as e:
+            missing_scipy = True
+
+        if (missing_scipy == False):

```

Comment:
this looks a bit like my DO_NOT_DEFINE_NDEBUG :-D

Reply:
DO_NOT_GET_NOT_JOKE :+1:

Reply:
https://www.youtube.com/watch?v=63NCvjX7X8E